packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -374,16 +374,34 @@ class LinearGradient extends Gradient {
       return b.scale(t);
     if (b == null)
       return a.scale(1.0 - t);
-    final _ColorsAndStops interpolated = _interpolateColorsAndStops(a.colors, a.stops, b.colors, b.stops, t);
+    final List<double> combinedStops = ((a._impliedStops() ?? <double>[0, 1]) + (b._impliedStops() ?? <double>[0, 1]))..sort();
```

Comment:
These 5 lines could probably be simplified by adding `(a._impliedStops() ?? <double>[0, 1])` and `(b._impliedStops() ?? <double>[0, 1])` to a `SplayTreeSet` [1]?

[1] https://master-docs.flutter.io/flutter/dart-collection/SplayTreeSet-class.html